### PR TITLE
A Faker test data generation tool with a simplistic performance test.

### PR DESCRIPTION
**Pull request by johrenberger**
While experimenting with Python and Faker I found that the test data samples could benefit from a more dynamic, scalable tool to more efficiently generate data on the fly.

csv_random_data_generator.py - This class has the Faker structure in place with several of the variables customized with AWS specific data. This lended to more realistic data generated at random. The bottom of the file includes several commands to generate an inital file (also attached) with 1000 records. There are opportunities to enhance it to include google or Azure options as well.

focus_validator_performance_test.py - Super simple performance test that exercises the 1,000 record csv to see what it's performance capabilities are. This can be enhanced for more complex scaling, created more for curiousity.

I am not a Python expert but thought the data generation would be useful for generating more interesting, on the fly, scaled data samples for testing.

Signed-off-by: Justin Ohrenberger [justinohrenberger@gmail.com](mailto:justinohrenberger@gmail.com)

### CODE REVIEW

**Review by johrenberger — COMMENTED** [review]
Signed-off-by: Justin Ohrenberger <justinohrenberger@gmail.com>

**Review by varunmittal91 — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Publish
+cdname: Publish
```
[the remark this answers is not in the document]
Comment (johrenberger) [reply .github/workflows/publish.yaml]: Looks like there was a mistake there. Thanks for catching.

File: tests/samples/csv_random_data_generator.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+       self.ServiceName = self.get_aws_service_name()
+       self.BilledCurrency = fake.currency_code()
+       self.BillingPeriodEnd = fake.date_time_this_month(before_now=True, after_now=False, tzinfo=None).isoformat()
+       self.BillingPeriodStart = fake.date_time_this_month(before_now=True, after_now=False, tzinfo=None).isoformat()
+       self.Region = self.get_aws_region()
+       self.ServiceCategory = self.get_aws_service_category()
+       self.ChargePeriodStart = fake.date_time_this_month(before_now=True, after_now=False, tzinfo=None).isoformat()
+       self.ChargePeriodEnd = fake.date_time_this_month(before_now=True, after_now=False, tzinfo=None).isoformat()
+       self.BilledCost = fake.pyfloat(left_digits=3,right_digits=2, positive=True)
+       self.AmortizedCost = fake.pyfloat(left_digits=3,right_digits=2, positive=True)
```
[the remark this answers is not in the document]
Comment (johrenberger) [reply tests/samples/csv_random_data_generator.py:27]: This was pulled from https://github.com/finopsfoundation/focus_validator/blob/main/tests/samples/all_pass.csv

This aligns with: https://github.com/finopsfoundation/focus_validator/blob/main/focus_validator/rules/base_rule_definitions/FV-M002-0003.yaml

Happy to change the name in my code but there may be another inconsistency to correct.

File: .gitignore
[the remark this answers is not in the document]
Comment (johrenberger) [reply .gitignore:1]: Agreed, file rolled back to original version.

**Pull request merged**